### DISCUSSION TITLE
Best way to create UI elements

### DISCUSSION THREAD

**Discussion by ghost** [picoe/Eto#2278] [Q&A] [answered]
Hey guys,

I'm having a bit of trouble figuring out what the best way to create child UI elements is, other than using the constructor. I'm hoping to avoid the constructor so that I can take advantage of properties set in the object initializer on instantiation. To do this, I tried using `OnLoad()` and `OnPreLoad()`, but neither of them seems to work properly - the callback is called on my `MainForm`, but not on the child UI elements (that are created in the MainForm callback). Anyone know how to properly do this the correct way (create UI automatically after the constructor)?

**Comment by cwensley** [comment]
OnLoad/PreLoad are only called when the control is added to a window that is shown.  They should theoretically work though as this where StackLayout and DynamicLayout creates their content.  If you have a concrete example of what you're trying to do maybe I can help further.

**Comment by ghost** [comment, marked as answer]
Haha my apologies, I realised where I had gone wrong. I was putting the call to base.OnXXX() (which recurses the event to child controls) before I added the child controls. Essentially, I updated the children before there were any, so the call went nowhere. Bit of an annoying thing to debug but it wasn't anything from you guys!
